library/spdm_crypt_lib/libspdm_crypt_cert.c

```diff
@@ -780,7 +780,78 @@ static bool libspdm_verify_leaf_cert_basic_constraints(const uint8_t *cert, size
 }
 
 /**
- * Verify leaf cert extend spdm OID
+ * Verify leaf cert spdm defined extended key usage
+ *
+ * @param[in]  cert                  Pointer to the DER-encoded certificate data.
+ * @param[in]  cert_size             The size of certificate data in bytes.
+ * @param[in]  is_requester_cert     Is the function verifying requester or responder cert.
+ *
+ * @retval  true   verify pass, two cases:
+ *                 1. spdm defined eku is not present in cert;
+ *                 2. spdm defined eku is compliant with requester/responder identity;
+ * @retval  false  verify fail, two cases:
+ *                 1. requester's cert has only responder auth oid in eku;
+ *                 2. responder's cert has only requester auth oid in eku;
+ **/
+static bool libspdm_verify_leaf_cert_spdm_eku(const uint8_t *cert, size_t cert_size,
```

Comment:
This will need a unit test.

Reply:
UT case and corresponding certs for testing have been added.